internal/idle/idle.go

```diff
@@ -16,41 +16,45 @@
  *
  */
 
-package grpc
+// Package idle contains a component for managing idleness (entering and exiting)
+// based on RPC activity.
+package idle
 
 import (
 	"fmt"
 	"math"
 	"sync"
 	"sync/atomic"
 	"time"
+
+	"google.golang.org/grpc/grpclog"
 )
 
 // For overriding in unit tests.
 var timeAfterFunc = func(d time.Duration, f func()) *time.Timer {
 	return time.AfterFunc(d, f)
 }
 
-// idlenessEnforcer is the functionality provided by grpc.ClientConn to enter
+// IdlenessEnforcer is the functionality provided by grpc.ClientConn to enter
```

Comment:
Should all of these names not include the `Idleness` prefix?

I'm ok with names being `idle.Enforcer`, `idle.Manager`, and `noopManager`, `managerImpl`, `idle.ManagerOptions` and `idle.NewManager()`.

If you feel that is not descriptive enough, we can rename the package `idleness` and then these names would be `idleness.Enforcer` and so on.

Wdyt?